src/app/ReadClient.cpp

```diff
@@ -527,7 +529,7 @@ CHIP_ERROR ReadClient::ProcessAttributeReportIBs(TLV::TLVReader & aAttributeRepo
                 attributePath.mListOp = ConcreteDataAttributePath::ListOperation::ReplaceAll;
             }
 
-            mpCallback.OnAttributeData(this, attributePath, &dataReader, statusIB);
+            mpCallback.OnAttributeData(this, &version, attributePath, &dataReader, statusIB);
```

Comment:
Yeah, see, this is passing a pointer into the stack of this function.  That is NOT a pointer we want callees saving.

Reply:
Just update with Optional<DataVersion>, thanks